### PR TITLE
Merging JSdoc setup from my branch feature/jsdoc-setup

### PR DESCRIPTION
**Pull request by andreGarvin**
@rdepena @datamadic @joneit @whyn07m3

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -73,21 +73,27 @@ module.exports = function(grunt) {
...
+
     grunt.registerTask('start-repl', function() {
         const finRepl = require(`./out/repl/index.js`);
         const done = this.async();
         finRepl.startRepl();
     });
-    
+
+    grunt.registerTask('doc', function() {
+        let exec = require('child_process').exec;
```
Comment (whyn07m3) [inline Gruntfile.js]: This should go to the top of the page and be `const exec = require('child_process').exec;`

File: tsconfig.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
         "declaration": true,
         "outDir": "out",
         "strictNullChecks": false,
-        "sourceMap": true
+        "sourceMap": true,
+        "removeComents": false
```
Comment (whyn07m3) [inline tsconfig.json]: shouldn't it be removeCo**mm**ents

File: Gruntfile.js (hunk 1 above)
Comment (andreGarvin) [inline Gruntfile.js]: Why would you write `const` instead of a `let`, if you are not changing the assignment in that callback ?

**Comment by rdepena** [comment]
@andreGarvin you need to remove the docs folder.

File: Gruntfile.js (hunk 1 above)
Comment (whyn07m3) [inline Gruntfile.js]: @andreGarvin `const` because you are not reassigning

File: jsdoc-conf.json [hunk 3]
```diff
@@ -11,7 +11,12 @@
         "monospaceLinks": false,
         "cleverLinks": false,
         "default": {
-            "outputSourceFiles": false,
+            "staticFiles": {
+	    	"include": [
+			"./jsdoc-template/static"
+		]
+	    },
+	    "outputSourceFiles": false,
```
Comment (whyn07m3) [inline jsdoc-conf.json]: indentations here are messed up a bit

**Comment by andreGarvin** [comment]
@whyn07m3 is that all that needs to be completed ?

**Comment by rdepena** [comment]
👍 will create tasks to fill in the docs.

**Pull request merged**